core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/GroupByConfig.java

```diff
@@ -0,0 +1,224 @@
+import co.cask.hydrator.plugin.batch.aggregator.function.Avg;
+import co.cask.hydrator.plugin.batch.aggregator.function.Count;
+import co.cask.hydrator.plugin.batch.aggregator.function.CountAll;
+import co.cask.hydrator.plugin.batch.aggregator.function.First;
+import co.cask.hydrator.plugin.batch.aggregator.function.Last;
+import co.cask.hydrator.plugin.batch.aggregator.function.Max;
+import co.cask.hydrator.plugin.batch.aggregator.function.Min;
+import co.cask.hydrator.plugin.batch.aggregator.function.Sum;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Config for group by types of plugins.
+ */
+public class GroupByConfig extends PluginConfig {
+  @Description("Comma separated list of record fields to group by. " +
+    "All records with the same value for all these fields will be grouped together. " +
+    "The records output by this aggregator will contain all the group by fields and aggregate fields. " +
+    "For example, if grouping by the 'user' field and calculating a count aggregate called 'numActions', " +
+    "output records will have a 'user' field and 'numActions' field.")
+  private final String groupByFields;
+
+  @Description("Aggregates to compute on grouped records. " +
+    "Supported aggregate functions are count, count(*), sum, avg, min, max, first, last. " +
+    "A function must specify the field it should be applied on, as well as the name it should be called. " +
+    "Aggregates are specified using syntax: \"name:function(field)[, other aggregates]\"." +
+    "For example, 'avgPrice:avg(price),cheapest:min(price)' will calculate two aggregates. " +
+    "The first will create a field called 'avgPrice' that is the average of all 'price' fields in the group. " +
+    "The second will create a field called 'cheapest' that contains the minimum 'price' field in the group")
+  private final String aggregates;
+
+  public GroupByConfig() {
+    this.groupByFields = "";
+    this.aggregates = "";
+  }
+
+  @VisibleForTesting
+  GroupByConfig(String groupByFields, String aggregates) {
+    this.groupByFields = groupByFields;
+    this.aggregates = aggregates;
+  }
+
+  Set<String> getGroupByFields() {
+    Set<String> fields = new HashSet<>();
+    for (String field : Splitter.on(',').trimResults().split(groupByFields)) {
+      fields.add(field);
+    }
+    if (fields.isEmpty()) {
+      throw new IllegalArgumentException("The 'groupByFields' property must be set.");
+    }
+    return fields;
+  }
+
+  List<FunctionInfo> getAggregates() {
+    List<FunctionInfo> functionInfos = new ArrayList<>();
+    Set<String> aggregateNames = new HashSet<>();
+    for (String aggregate : Splitter.on(',').trimResults().split(aggregates)) {
+      int colonIdx = aggregate.indexOf(':');
```

Comment:
if it doesn't match the regex, you don't really know why, you just know it didn't match.